### PR TITLE
[Arrow] Add filesystem feature

### PR DESCRIPTION
**Pull request by isaacbrodsky**
**Describe the pull request**
Arrow has optional features for working with filesystems. This PR adds an optional `filesystem` feature for generic (local) filesystem support. S3 and HDFS aren't added as options but could be with the appropriate dependencies.

- What does your PR fix? Fixes issue #

- Which triplets are supported/not supported? Have you updated the CI baseline?
No change expected.

- Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?
Yes

### CODE REVIEW

File: ports/arrow/CONTROL [hunk 1]
```diff
@@ -4,7 +4,7 @@ Build-Depends: boost-system, boost-filesystem, boost-multiprecision, boost-algor
 Homepage: https://github.com/apache/arrow
```
Comment (ras0219-msft) [inline ports/arrow/CONTROL:4]: Could you please increment the `Version:` field, since this is a meaningful change to the port?

**Review by ras0219-msft — DISMISSED** [review]

File: ports/arrow/CONTROL (hunk 1 above)
Comment (isaacbrodsky) [reply ports/arrow/CONTROL:4]: Oops, missed that. Updated now.

**Comment by strega-nil** [comment]
Looks good, thanks @isaacbrodsky :)

**Pull request merged**